pkg/operator/ceph/object/objectstore.go

```diff
@@ -289,6 +290,21 @@ func createSimilarPools(context *Context, pools []string, poolSpec model.Pool) e
 			if err != nil {
 				return errors.Errorf("failed to create pool %s for object store %s", name, context.Name)
 			}
+		} else {
+			logger.Infof("pool %s already exists", name)
+			if !isECPool {
+				//detect if the replication is different from the pool details
+				replicationChanged := false
+				if poolDetails.Size != poolSpec.ReplicatedConfig.Size {
+					replicationChanged = true
+				}
+				if replicationChanged {
```

Comment:
How about if you check directly for `if poolDetails.Size != poolSpec.ReplicatedConfig.Size`? No need for the `replicationChanged` variable.

Reply:
I think that `replicationChanged` improves the redability?.

Reply:
If the conditions were more complex I would agree, but the single condition is simple enough

Reply:
Done , please review now.
Thanks